.github/workflows/iroha2-dev-pr.yml

```
@@ -40,44 +40,38 @@ jobs:
     container:
       image: hyperledger/iroha2-ci:nightly-2024-01-12
     steps:
-      - uses: actions/checkout@v3
-      # TODO Remove this step #2165
-      # - name: Adjust toolchain
-      #   run: |
-      #     rustup component add llvm-tools-preview
+      - uses: actions/checkout@v4
       - uses: Swatinem/rust-cache@v2
       - name: Run tests, with coverage
-        run: |
-          mold --run cargo llvm-cov clean --workspace
-          mold --run cargo llvm-cov --doc --no-report --all-features --workspace --no-fail-fast --ignore-filename-regex='(^client_cli/|main\.rs)'
-          mold --run cargo llvm-cov --no-report --ignore-filename-regex='(^client_cli/|main\.rs)' --all-features --workspace --no-fail-fast
+        run: mold --run cargo test --all-features --workspace --no-fail-fast
+        env:
+          RUSTFLAGS: "-C instrument-coverage"
+          LLVM_PROFILE_FILE: "iroha-%p-%m.profraw"
       - name: Generate lcov report
-        run: |
-          # generate report without tests
-          # https://github.com/taiki-e/cargo-llvm-cov#merge-coverages-generated-under-different-test-conditions
-          mold --run cargo llvm-cov --doctests --no-run --all-features --workspace --lcov --output-path lcov.info
+        if: always()
+        run: grcov . --binary-path ./target/debug/ -s . -t lcov --branch --ignore-not-existing --ignore "/client_cli/*|**/main.rs" -o lcov.info
```

The same as above about `cargo-test`.

---

I don't think `--ignore` option allows `|` pattern as it accepts not a _regex_ but a _glob_ pattern.
Instead, to have multiple options will be a right way, which seems to be a common pattern:

```suggestion
        run: grcov . --binary-path ./target/debug/ -s . -t lcov --branch --ignore-not-existing --ignore "/client_cli" --ignore "**/main.rs" -o lcov.info
```

A further rationale is:
- the [interface] follows the [clap::ArgAction::Append] specification
- the [conditions] are [combined by OR]

[interface]: https://github.com/mozilla/grcov/blob/cc77ce34164fc3ea80ac579d1c15f36c9734133c/src/main.rs#L199-L200
[clap::ArgAction::Append]: https://docs.rs/clap/latest/clap/_derive/_tutorial/chapter_2/index.html#positionals
[conditions]: https://github.com/mozilla/grcov/blob/cc77ce34164fc3ea80ac579d1c15f36c9734133c/src/path_rewriting.rs#L262-L264
[combined by OR]: https://docs.rs/globset/latest/globset/struct.GlobSet.html#method.is_match